src/views/UserSearch/UserSearch.js

```diff
@@ -520,15 +526,40 @@ class UserSearch extends React.Component {
 
     try {
       this.context.sendCallout({ message: <FormattedMessage id="ui-users.reports.inProgress" /> });
-      const { reportData } = await cashDrawerReport.POST(reportParameters);
+      const reportData = await cashDrawerReport.POST(reportParameters);
 
       if (isEmpty(reportData)) {
         this.context.sendCallout({
           type: 'error',
           message: <FormattedMessage id="ui-users.reports.noItemsFound" />,
         });
       } else {
-        // TODO: form the report depends on format
+        const servicePoints = currentUser.servicePoints;
+        const chosenServicePoint = servicePoints.find(sp => sp.id === servicePoint);
+        const headerData = {
+          ...reportParameters,
+          createdAt: chosenServicePoint?.name ?? '',
+          sources: reportParameters.sources.join(', '),
+        };
+
+        const report = new CashDrawerReconciliationReportPDF({
+          data: reportData,
+          intl,
+          headerData,
+        });
+
+        switch (format) {
```

Comment:
@annamelnyk do you think we could avoid the switch here? 
It looks like the only case when the pdf is not generates is when the csv is chosen. So perhaps a simple if/else could work well here?

````js
if (format !== 'csv') {
  report.toPDF();
}
````

Reply:
Thanks for review @mkuklis, but this switch statement not fully ready yet, I have one more story I currently working on, there will be the same report but in `csv` format. So `switch` will have more statements but anyway I will think if I can simplify the statement